lib/rules/no-error-cause.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * @author Sosuke Suzuki <https://github.com/sosukesuzuki>
+ * See LICENSE file in root directory for full license.
+ */
+
+"use strict"
+
+const { CONSTRUCT, READ, ReferenceTracker } = require("eslint-utils")
+
+/**
+ * @typedef {import("estree")} Node
+ * @typedef {import("estree").ClassExpression | import("estree").ClassDeclaration} ClassNode
+ * @typedef {import("estree").CallExpression} CallExpression
+ */
+
+const traceMap = {}
+for (const errorConstructorName of ["Error", "AggregateError"]) {
+    traceMap[errorConstructorName] = { [CONSTRUCT]: true, [READ]: true }
+}
+
+/**
+ * @param {Node} node
+ * @returns {boolean}
+ */
+function isSuperCall(node) {
+    return node.type === "CallExpression" && node.callee.type === "Super"
+}
+
+/**
+ * Checks if the received node is a construcotr call with cause option.
+ * e.g. `new Error("message", { cause: foo })`, `super("message", { cause: foo })`
+ *
+ * @param {Node} node
+ * @returns {boolean}
+ */
+function isConstructCallWithCauseOption(node) {
+    if (node.type !== "NewExpression" && !isSuperCall(node)) {
+        return false
+    }
+    const secondArg = node.arguments[1]
```

Comment:
Should make sure the first argument is not a `SpreadElement` `new Error(...foo, {cause})`